src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    if (!Animal.alive) {
+      Animal.alive = [];
+    } else {
+      Animal.alive.push(this);
+    }
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.isCarnivore = true;
+  }
+
+  bite(animal) {
+    if (!animal.hidden && !animal.isCarnivore) {
+      animal.health -= 50;
+    }
+
+    Animal.alive = Animal.alive.filter(an => an.health > 0);
```

Comment:
you will do filter method even if all animals is alive
It is better to do this only if you need - please think about it